### PR TITLE
Created structure and set configuration for redux

### PR DESCRIPTION
**Pull request by tomaszprc**

### CODE REVIEW

File: src/redux/projects/types.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  title: string;
+  description: string;
+  timeTrackerIds: number[];
+}
+
+export interface ProjectStateType {
+  projectsList: Project[];
+}
+
+export const ADD_PROJECT = "ADD_PROJECT";
```
Comment (korczas) [inline src/redux/projects/types.ts:12]: keep constants on top of the file

use `domain/action` convention, eg `PROJECTS/ADD_PROJECT`

File: src/redux/projects/types.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+export interface AddProjectAction {
+  type: typeof ADD_PROJECT;
+  payload: Project;
+}
+
+export type ProjectActionTypes = AddProjectAction;
```
Comment (korczas) [inline src/redux/projects/types.ts:19]: what was the idea behind this export?

File: src/pages/Projects/ProjectPage/ProjectPage.tsx [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+}
+
 const ProjectPage = ({}: ProjectPageProps) => {
-  return <div>Project page</div>;
+  let { id } = useParams<ProjectParams>();
+
+  return (
+    <Container>
+      <div>{id}</div>
+    </Container>
```
Comment (korczas) [inline src/pages/Projects/ProjectPage/ProjectPage.tsx:16]: no need to use Container

File: src/redux/store.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { combineReducers, createStore } from "redux";
+import { composeWithDevTools } from "redux-devtools-extension";
+import { projectReducer } from "./projects/reducers";
+
+const rootReducer = combineReducers({
+  projects: projectReducer,
+});
+
+export const store = createStore(rootReducer, composeWithDevTools());
```
Comment (korczas) [inline src/redux/store.tsx:9]: you are missing type export

```
export type RootState = ReturnType<typeof rootReducer>
```

File: src/pages/Projects/Projects.tsx [hunk 5]
```diff
@@ -2,38 +2,27 @@ import React, { useState } from "react";
...
-      timeTrackIds: [2, 3],
-    },
-    {
-      id: 2,
-      title: "Tytul projektu 2",
-      description: "Opis projektu 2",
-      timeTrackIds: [4, 2],
-    },
-  ]);
+  const projectsList = useSelector((state: any) => state.projects.projectsList);
```
Comment (korczas) [inline src/pages/Projects/Projects.tsx:17]: 1. don't use `any`. Never ever. If you are using `any` then there is no sense of using typescript. 
2. state type should be imported from redux/store.ts file (see related comment)
3. `(state: RootState) => state.projects.projectsList` should be moved to selectors (redux/projects/selectors.tsx)

**Review by korczas — COMMENTED** [review]
for reducer folder you can use convention:
actions.ts - action for that reducer
reducer.ts - reducer
types.ts - consts and types. Consts should be grouped on top of the file
selectors.ts - (optional) if you have any selector, than put it here. Having selectors is a good practice ;) 
index.ts - (optional) only for exporting everything, eg.
```
export * from './actions';
export * from './reducer';
export * from './types';
export * from './selectors';
```
This simplifies importing later, cuz you can aggregate imports in one statement
```
import {getProjects, addProject, ProjectType} from 'redux/projects'
```
Keep in mind that it works only with named exports, so if you have default, they won't be available.

File: src/redux/projects/types.ts (hunk 2 above)
Comment (tomaszprc) [reply src/redux/projects/types.ts:19]: The Idea is from https://redux.js.org/recipes/usage-with-typescript#type-checking-actions--action-creators , in future when i have more interfaces for Actions I can provide interface by one variable. I mean this line on the linked page: export type ChatActionTypes = SendMessageAction | DeleteMessageAction , but I'm not sure what if we have more Actions e.g 30, probably our code could be messy, what do you think?

File: src/pages/Projects/ProjectPage/ProjectPage.tsx (hunk 3 above)
Comment (tomaszprc) [reply src/pages/Projects/ProjectPage/ProjectPage.tsx:16]: will fix in another task

**Pull request merged**